pyOCD/tools/gdb_server.py

```diff
@@ -65,6 +65,7 @@ def build_parser(self):
         parser.add_argument("-t", "--target", dest="target_override", choices=supported_targets, default=None, help="Override target to debug.  Supported targets are: " + ", ".join(supported_targets), metavar="TARGET")
         parser.add_argument("-n", "--nobreak", dest="break_at_hardfault", default=True, action="store_false", help="Disable halt at hardfault handler.")
         parser.add_argument("-r", "--reset-break", dest="break_on_reset", default=False, action="store_true", help="Halt the target when reset.")
+        parser.add_argument("-C", "--vector-catch", default='', help="Select enabled vector catch options, one letter per enabled source in any order. (h=hard fault, b=bus fault, m=mem fault, i=irq err, s=state err, c=check err, p=nocp, r=reset, a=all, n=none)")
```

Comment:
Right. It really just ignores any character it doesn't recognize, so for example 'z' is also equivalent. The relevant code that converts from the string to mask is in `GDBServer.__init__()`:

```c
        mask = ((Target.CATCH_HARD_FAULT if ('h' in self.vector_catch or self.break_at_hardfault) else 0) \
                | (Target.CATCH_BUS_FAULT if 'b' in self.vector_catch else 0) \
                | (Target.CATCH_MEM_FAULT if 'm' in self.vector_catch else 0) \
                | (Target.CATCH_INTERRUPT_ERR if 'i' in self.vector_catch else 0) \
                | (Target.CATCH_STATE_ERR if 's' in self.vector_catch else 0) \
                | (Target.CATCH_CHECK_ERR if 'c' in self.vector_catch else 0) \
                | (Target.CATCH_COPROCESSOR_ERR if 'p' in self.vector_catch else 0) \
                | (Target.CATCH_CORE_RESET if ('r' in self.vector_catch or self.break_on_reset) else 0) \
                | (Target.CATCH_ALL if 'a' in self.vector_catch else 0))
```